greenplumpython/experimental/embedding.py

```diff
@@ -136,8 +144,10 @@ def create_index(self, column: str, model_name: str) -> gp.DataFrame:
                 distribution_type="hash",
             )
             .check_unique(self._dataframe.unique_key)
-            .create_index(columns={embedding_col_name}, method="ivfflat")
         )
+        if method is not None:
+            assert method == "hnsw" or method == "ivfflat"
```

Comment:
since there may be more method , `assert method in ["ivfflat", "hnsw"]` may be bettor?